### PR TITLE
[NUI] Change TouchArea to TouchAreaOffset

### PR DESCRIPTION
**Pull request by JoogabYun**
Change TouchArea operation

You can set offset of the touch area.

for example)
  View view = new View();
  view.Size = new Size(10, 10);
  view.TouchEvent += OnTouch;
  view.SetTouchAreaOffset(-100, 100, 100, -100); // left, right, bottom, top

then touch area is 210x210.
this is view.width + TouchAreaOffset.right - TouchAreaOffset.left
and view.height + TouchAreaOffset.bottom -TouchAreaOffset.top

### API Changes ###

 - ACR:

### CODE REVIEW

**Review by huiyueun — APPROVED** [review]
LGTM

**Review by everLEEst — COMMENTED** [review]
Looks good to me but someone else need to approve.

**Comment by dongsug-song** [comment]
@JoogabYun 
could you update the csharp-binder's patch on this PR? and csharp-binder patch should be merged prior to this PR. otherwise there can be a runtime exception which blocks the release of tizenfx by TCT hurdle

**Comment by JoogabYun** [comment]
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/256790/

I will merge this NUI patch after the csharp binder is merged.

File: src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs:1295]: Is it not possible to extend like a margin?
If I want to extend touch area just right size, could I do?
if it possible `Extents` more proper

Comment (JoogabYun) [reply src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs:1295]: You can only extend one of left, right, bottom and top.
This is based on the top left x, y coordinates.

These coordinates can have negative value. so I can not using 'Extents'.
because 'Extents' is unsigned int value.

Comment (myroot) [reply src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs]: How to use to extend 10 pixel for 4 direction (left, right, bottom, top)?
image

1. new Rectangle(-10, 10, 10, -10)
2. new Rectangle(-10, View.Size.Width + 10, View.Size.Height + 10, -10);

And there is no constructor to accept left, right, bottom, top

Comment (JoogabYun) [reply src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs]: 1. new Rectangle(-10, 10, 10, -10)
This is why I added an 'offset' to the name. 

 there is no constructor to accept left, right, bottom, top
--> sorry,  please explain what you are saying.

Comment (myroot) [reply src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs]: Do not use `Rectangle` class, it not proper to represent offset.
How about think about it?
``` c# 
var is_it_a_width_of_touch_area = view.TouchAreaOffset.Width ; // ???
```
>there is no constructor to accept left, right, bottom, top

See it, no constructor that has arguments as `(int left, int right, int top, int down)`
https://github.com/Samsung/TizenFX/blob/c6147db7131e1b759a1d2da515df3d3af361426a/src/Tizen.NUI/src/public/Common/Rectangle.cs#L34-L74

I recommend to extend `Vector4` class to represent four direction offset

Comment (JoogabYun) [reply src/Tizen.NUI/src/public/BaseComponents/ViewEvent.cs]: Oh I see, I will consider it.

**Comment by JoogabYun** [comment]
conflicts 발생으로 https://github.com/Samsung/TizenFX/pull/2948로 패치를 다시 올립니다.

**Pull request closed without merging**
